lib/common/exceptions/detailed_schema_errors.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'common/exceptions/base_error'
+require 'common/exceptions/serializable_error'
+
+module Common
+  module Exceptions
+    class DetailedSchemaErrors < BaseError
+      # Expects array of JSONSchemer errors
+      def initialize(raw_schema_errors)
+        @raw_errors = raw_schema_errors
+        raise TypeError, 'the resource provided has no errors' if raw_schema_errors.blank?
+      end
+
+      def errors
+        @raw_errors.map do |raw_error|
+          type = raw_error['type'].downcase
```

Comment:
Is the list of possible types documented?

Reply:
I think the fallback I just added should be self-documenting enough show what's supported, but I can add in a constant to check instead if you think that'd make things clearer.